### ISSUE TITLE
[documentation] update Data Types ROW doc to include delimited flag

### ISSUE THREAD

**Issue by steveburnett** [prestodb/presto#21897]
Update https://prestodb.io/docs/current/language/types.html#row to document the Boolean delimited flag. 

## Expected Behavior or Use Case
The Boolean delimited flag should be documented on https://prestodb.io/docs/current/language/types.html#row. 

## Presto Component, Service, or Connector
Documentation.

## Possible Implementation
Update https://github.com/prestodb/presto/blob/master/presto-docs/src/main/sphinx/language/types.rst#row with the requested content. 

## Context
I thank @mbasmanova for identifying this lack of documentation in [their comment reviewing PR 21869](https://github.com/prestodb/presto/pull/21869#issuecomment-1935573856).

**Comment by tdcmeehan** [comment]
I don't think that the delimited flag should be documented on the row type itself.  The reason is, it's not directly exposed to the user.  Delimited indicates if a row's fields are case sensitive, and I think actually better documentation should go around case sensitivity support in Presto.
